### ISSUE TITLE
This project lacks automated testing

### ISSUE THREAD

**Issue by YassineLambarki** [SkyCascade/SkyLearn#3]
**Is your feature request related to a problem? Please describe.**
Difficult to test overall stability after a feature addition.

**Describe the solution you'd like**
To make adding features and bugfixes error-prone, this project needs automated testing or unit tests.
This project is at a maturity level that seriously needs a testing solution, especially to accept new features from different developers.

**Comment by adilmohak** [comment]
@YassineLambarki Good point, we're working on it. Also feel free to contribute to that (adding test cases).

**Comment by adilmohak** [comment]
> **Is your feature request related to a problem? Please describe.** Difficult to test overall stability after a feature addition.
> 
> **Describe the solution you'd like** To make adding features and bugfixes error-prone, this project needs automated testing or unit tests. This project is at a maturity level that seriously needs a testing solution, especially to accept new features from different developers.

Now automated testing has been integrated into GitHub actions. Although, obviously we have to add more test cases.
